### PR TITLE
Serialize JSON errors and headers JSON support

### PR DESCRIPTION
**Pull request by sfc-gh-pkommini**
1. Fix JSON datetime serialize error
2. Add support for JSON serialized headers

### CODE REVIEW

File: lambda_src/drivers/process_https.py [hunk 1]
```diff
@@ -44,14 +44,17 @@ def process_row(
...
     req_kwargs = parse_header_dict(kwargs)
-
-    req_headers = {
-        k: v.format(**req_kwargs) for k, v in parse_header_dict(headers).items()
-    }
-    req_headers.setdefault('User-Agent', 'Snowflake Generic External Function 1.0')
+    req_headers = (
+        loads(headers)
+        if headers.startswith('{')
+        else {k: v.format(**req_kwargs) for k, v in parse_header_dict(headers).items()}
```
Comment (sfc-gh-afedorov) [inline lambda_src/drivers/process_https.py:52]: I think we want `v.format(**req_kwargs)` to apply to both `parse_header_dict` and `loads` results

File: lambda_src/drivers/process_https.py [hunk 2]
```diff
@@ -44,14 +44,20 @@ def process_row(
             raise RuntimeError('url must start with https://')
     else:
         req_host = base_url
-        req_path = url or '/'
```
Comment (sfc-gh-afedorov) [inline lambda_src/drivers/process_https.py:47]: not a blocker to merge, but I think we either missed or mis-merged some changes w.r.t. how `base_url` and `url` work together -- I think some unit tests would help here.

**Review by sfc-gh-afedorov — APPROVED** [review]

File: lambda_src/drivers/process_https.py [hunk 3]
```diff
@@ -95,7 +103,9 @@ def process_row(
     next_url: Optional[str] = req_url
     row_data: List[Any] = []
 
+    LOG.debug(f'Headers: {req_headers}')
```
Comment (sfc-gh-afedorov) [inline lambda_src/drivers/process_https.py:106]: this will log secrets in prod

**Review by sfc-gh-afedorov — CHANGES_REQUESTED** [review]

**Review by sfc-gh-afedorov — APPROVED** [review]

File: lambda_src/drivers/process_https.py (hunk 2 above)
Comment (sfc-gh-pkommini) [reply lambda_src/drivers/process_https.py:47]: @sfc-gh-afedorov Good catch raised another PR for it.

**Pull request merged**